### ISSUE TITLE
AI labour shortage/ AI don't rebuild houses

### ISSUE THREAD

**Issue by PitchNeeded** [Sh0wdown/UnofficialCrusaderPatch#131]
The ox tether spam bug may be fixed now, but AIs can still have labour shortages.

I think this might be connected with AIs not always rebuilding destroyed houses I noticed in a match recently that The Emir didn't rebuild his 3 houses at the side of his castle that were destroyed by Richard's catapults. Even after he had destroyed Richard's catapults with Arabian archers, the houses were not rebuilt. Therefore he had a labour shortage, which meant there no were ale buildings with workers and some of his quarries with no workers, so that weakened his economy. There were no troops nearby that would have and he had enough gold to buy wood (and eventually gathered some wood too) so I don't know why the houses were not rebuilt.  

I notice sometimes other large population AIs - Saladin and Abbot particularly, sometimes have labour shortages and I am thinking this may be the same issue.

**Comment by Heroesflorian** [comment]
I think this is mostly a result of the AI prioritising recruiting over occupying buildings with workers. 
Some AI with high recruiting speed can turn all newly-spawned peasants into soldiers before they can turn into workers. This can lead to buildings not getting workers temporarily, until the "recruiting session" is over.

**Comment by PitchNeeded** [comment]
Yes and this especially happens if their farms are frequently under pressure from enemy missile troops as they will have to keep recruiting archers/crossbowmen and/or or counter attack melee units to send out to try and counter the enemy units.

**Comment by Heroesflorian** [comment]
Yeah, I do not see this as an issue related to houses or general lack of population, though. Rather a possibly wrong prioritization of existing (sufficient) population. But then again, maybe troops are more effective to defend the remaining workers than sending new workers out that would get shot again immediately as well... 

Nonetheless, the peasant spawning speed depends on popularity. And this - as discussed in another issue - is indeed something the AI could improve upon - i.e. no longer dropping down to 50 popularity, but instead manage it between e.g. 80-100 with no loss of tax efficiency but considerably higher average peasant spawning speed.

**Comment by Krarilotus** [comment]
you can change the Tax rates now. 
set LowestPopularity to a high numer like 8500
and set HighestPopularity to sth like 9700 and he has a good tax income and a good peasant income.

Now there is still the bug, that if the AI has 21 free citizens, it will destroy a house, making it have overpopulation malus. Which is totally unecessary. I wish for that the AI woud only demolish houses, if it has theoretically 32 free citizens (it usually calculates: (total pop) - (work force needed)), then it is already capped at 24, since that is the maximum of population waiting at the fire camp, and it wont suffer from the overpopulation malus. I also havnt found a value in the AIC that is related to that 20 maximum free population.... Maybe you are able to figure that out?
